src/main/java/com/kodilla/ecommercee/domain/Cart.java

```diff
@@ -1,4 +1,63 @@
 package com.kodilla.ecommercee.domain;
 
+import com.kodilla.ecommercee.exception.ProductNotFoundInCartException;
+
+import javax.persistence.Entity;
+import javax.persistence.Table;
+import javax.persistence.GeneratedValue;
+import javax.persistence.ManyToMany;
+import javax.persistence.GenerationType;
+import javax.persistence.Id;
+import javax.persistence.CascadeType;
+import javax.persistence.JoinTable;
+import javax.persistence.JoinColumn;
+import javax.validation.constraints.NotNull;
+import java.util.ArrayList;
+import java.util.List;
+
+@Entity
+@Table(name="CARTS")
 public class Cart {
+    private Long id;
+    private List<Product> products = new ArrayList<>();
+
+    public Cart() {};
+
+    public Cart(Long id, List<Product> productList) {
+        this.id = id;
+        this.products = productList;
+    }
+
+    @Id
+    @NotNull
+    @GeneratedValue(strategy = GenerationType.AUTO)
+    public Long getId() {
+        return id;
+    }
+
+    @ManyToMany(cascade = CascadeType.ALL)
+    @JoinTable(
+            name = "JOIN_CART_PRODUCT",
+            joinColumns = {@JoinColumn(name = "PRODUCT_ID", referencedColumnName = "PRODUCT_ID")},
+            inverseJoinColumns = {@JoinColumn(name = "CART_ID", referencedColumnName = "CART_ID")}
+    )
+    public List<Product> getProductList() {
+        return products;
+    }
+
+    public void addToCart(Product product) {
+        products.add(product);
+    }
+
+    public void removeFromCart(Product product) {
+        if(products.contains(product)) {
+            products.remove(product);
+        } else {
+            throw new ProductNotFoundInCartException("Product was not found in the cart");
+        }
+    }
+
```

Comment:
To można by zdefiniować w Controllerze, lub w klasie service po zapytaniu z endpointu, żeby sprawdził czy istnieje dany obiekt od razu..

Reply:
Ok, możemy to przerzucić na poziom Controllera, tylko żeby o tym pamiętać. Zawarłem to u siebie po prostu po to, żeby się upewnić, że w mojej części kodu nie dochodzi do operacji mogących powodować błąd

Reply:
ale to trzeba będzie przerzucić finalnie mimo wszystko. Wyjdzie z tego koniec końcem martwy kod.